integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystemOpts.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.fuse;
+
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is an object containing all fuse-related options that
+ * either can be configured through command line when launching Alluxio fuse,
+ * or can be passed to Libfuse,
+ * or both.
+ */
+public final class AlluxioFuseFileSystemOpts {
+  private final String mAlluxioPath;
+  private final List<String> mFuseOptions;
+  private final boolean mIsDebug;
+  private final String mMountPoint;
+
+  private AlluxioFuseFileSystemOpts(
+      String alluxioPath, String mountPoint, List<String> fuseOptions, boolean isDebug) {
+    mAlluxioPath = alluxioPath;
+    mMountPoint = mountPoint;
+    mFuseOptions = fuseOptions;
+    mIsDebug = isDebug;
+  }
+
+  /**
+   * Constructs an AlluxioFuseFileSystemOpts with only Alluxio cluster configuration.
+   *
+   * @param conf Alluxio cluster configuration
+   * @return AlluxioFuseFileSystemOpts
+   */
+  public static AlluxioFuseFileSystemOpts create(AlluxioConfiguration conf) {
```

Comment:
Usually we put the `static create` methods before the `private constructor` methods
same for other